packages/curve-ui-kit/src/shared/ui/Metric.tsx

```diff
@@ -0,0 +1,193 @@
+  value: number
+  /** Optional formatter for metric value */
+  formatter?: (value: number) => string
+  /** If the value should be abbreviated to 1.23k or 3.45m */
+  abbreviate?: boolean
+  /** The number of decimals the value should contain */
+  decimals?: number
+  /** A unit can be a currency symbol or percentage, prefix or suffix */
+  unit?: Unit
+
+  /** Optional value that denotes a change in metric value since 'last' time */
+  change?: number
+
+  /** Notional values give extra context to the metric, like underlying value */
+  notional?: number
+  notionalFormatter?: (value: number) => string
+  notionalAbbreviate?: boolean
+  notionalDecimals?: number
+  notionalUnit?: Unit
+
+  size?: keyof typeof MetricSize
+  alignment?: Alignment
+  loading?: boolean
+}
+
+export const Metric = ({
+  label,
+
+  value,
+  formatter = (value: number) => formatValue(value, decimals),
+  abbreviate,
+  unit,
+  decimals = 1,
+
+  change,
+
+  notional,
+  notionalFormatter = (value: number) => formatNotionalValue(value, notionalDecimals),
+  notionalAbbreviate,
+  notionalUnit,
+  notionalDecimals,
+
+  size = 'medium',
+  alignment = 'start',
+  loading = false,
+}: Props) => {
+  unit = typeof unit === 'string' ? UNIT_MAP[unit as keyof typeof UNIT_MAP] : unit
+  const valueAbbr = abbreviate ?? unit?.abbreviate ?? false
+
+  notionalUnit = typeof notionalUnit === 'string' ? UNIT_MAP[notionalUnit as keyof typeof UNIT_MAP] : notionalUnit
+  const notionalAbbr = notionalAbbreviate ?? notionalUnit?.abbreviate ?? false
+
+  return (
+    <Box display="flex" flexDirection="column" alignItems={alignment} gap={Spacing.xs}>
+      <Typography variant="bodyXsRegular" color="textTertiary">
+        {label}
+      </Typography>
+
+      {loading ? (
+        <Skeleton
+          variant="text"
+          width="100%"
+          sx={{
+            fontSize: TYPOGRAPHY_VARIANTS[MetricSize[size]],
+          }}
+        />
+      ) : (
+        <Box display="flex" alignItems="baseline" gap={Spacing.xxs}>
+          {unit?.position === 'prefix' && (
+            <Typography variant={MetricSize[size]} color="textSecondary">
+              {unit.symbol}
+            </Typography>
+          )}
+
+          <Typography variant={MetricSize[size]} color="textPrimary">
+            {formatter(valueAbbr ? abbr(value) : value)}
+          </Typography>
+
+          {valueAbbr && (
+            <Typography variant={MetricSize[size]} color="textPrimary" textTransform="capitalize">
+              {suffix(value)}
+            </Typography>
+          )}
+
+          {unit?.position === 'suffix' && (
+            <Typography variant={MetricSize[size]} color="textSecondary">
+              {unit.symbol}
+            </Typography>
+          )}
+
+          {(change || change === 0) && (
+            <Typography
+              variant="highlightM"
+              color={change > 0 ? 'success' : change < 0 ? 'error' : 'textHighlight'}
```

Comment:
nb. there is a problem with the feedback colors: https://www.notion.so/Fix-feedback-colors-15b599aae06480b4a40ac99c6c163726?pvs=4